### PR TITLE
[Merged by Bors] - feat: Add `smdk` install support in CLI and release

### LINKED ISSUE #2644: CLI installation of SmartModule Development Kit - `smdk`
Users will install by running the fluvio install command:

`fluvio install smdk`

The expectation is this will install as `~/.fluvio/bin/smdk`.

---

This needs to be added into the publish and release workflows. It should also respect channels, but that's mostly on the Fluvio CLI to install into the correct extensions directory.

### PR DESCRIPTION
**Pull request by tjtelan**
Closes #2644 

Adds a name check to `fluvio install`.
If it starts with `fluvio-`, it is an extension and gets installed in the channel extension dir. Otherwise, it gets installed in the fluvio bin dir.

The result of `fluvio install smdk --develop` installs the latest published `smdk` to install in my fluvio path.

--

Added `smdk` to CI for build and publish for CI

Moving a lot of publish and release workflows into Makefile so I can test it easier locally

Got rid of the `cargo make` configs since they are out of date and unmaintained

### CODE REVIEW

File: crates/smart-module-development-kit/Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [package]
-name = "smdk"
+name = "smart-module-development-kit"
```
Comment (sehz) [inline crates/smart-module-development-kit/Cargo.toml:2]: Let's change to `smartmodule-development-kit`.   
We should change other smarmodule crates as well.

**Review by sehz — CHANGES_REQUESTED** [review]
Looks very good.  One minor not about naming

**Review by sehz — APPROVED** [review]
LGTM

**Comment by tjtelan** [comment]
Bors failed because while we had steps to upload SMDK in CI, but we weren't building it in the first place. Hopefully this will fix.

bors r+

**Comment by tjtelan** [comment]
Based on cranelift-codegen's build, I removed the arm and arm7 targets. I hope this build is the one!

https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/codegen/build.rs#L174

bors r+

**Comment by tjtelan** [comment]
This time it was publish to Github Release (`dev`) that failed. I checked on the Publish, and Release workflows to ensure that we don't have this kind of artifact missing traps waiting for us there.

bors r+

**Comment by tjtelan** [comment]
I really wish Bors supported rerunning jobs...

bors r+

**Comment by tjtelan** [comment]
I had to rebase after the last PR, but here we go again

bors r+

**Comment by tjtelan** [comment]
Removed Windows targets for `smdk` for now

bors r+

**Pull request closed without merging**